### PR TITLE
Removed null year manufactured check

### LINKED ISSUE #443: Empty Year Manufactured causes Add Device Failure
References user story #278 

On the add device page, if no year manufactured is specified the add device api call will fail. This is because of the `CheckNullArgument()` call on it in the `GetBsonDocument()` Device method. It is failing because we need to default the field to a null value because it is an int unlike the other fields. The fix is to simply remove the checknullargument on this field.

### PR DESCRIPTION
**Pull request by Kenny-Miller**
Closes #443

### CODE REVIEW

**Review by josephmoore99 — DISMISSED** [review]
We could also add a test for this if you got the time.

**Review by AshlynAdamson — APPROVED** [review]

**Comment by AshlynAdamson** [comment]
I think this is our tiniest PR yet lol

**Review by josephmoore99 — APPROVED** [review]

File: BEIMA.Backend.Test/DeviceFunctions/AddDeviceTest.cs [hunk 1]
```diff
@@ -80,6 +80,65 @@ public async Task NoDevice_AddDevice_ReturnsValidId()
...
+                deviceId = ((ObjectResult)await AddDevice.Run(request, logger)).Value?.ToString();
+            }
+
+            // ASSERT
+            Assert.DoesNotThrow(() => mockDb.Verify(mock => mock.GetDeviceType(It.IsAny<ObjectId>()), Times.Once));
+            Assert.DoesNotThrow(() => mockDb.Verify(mock => mock.InsertDevice(It.IsAny<BsonDocument>()), Times.Once));
+
+            Assert.IsNotNull(deviceId);
+            Assert.DoesNotThrow(() => mockStorage.Verify(mock => mock.PutFile(It.IsAny<IFormFile>()), Times.Once));
+            Assert.DoesNotThrow(() => mockDb.Verify(mock => mock.InsertDevice(It.IsAny<BsonDocument>()), Times.Once));
```
[the remark this answers is not in the document]
Comment (Kenny-Miller) [reply BEIMA.Backend.Test/DeviceFunctions/AddDeviceTest.cs:138]: Thats what I get for copy/pasting :/ . Looks like the above and below test also do that.

[the remark this answers is not in the document]
Comment (josephmoore99) [reply BEIMA.Backend.Test/DeviceFunctions/AddDeviceTest.cs:134]: Good that we're catching this in the other tests as well then.

[the remark this answers is not in the document]
Comment (Kenny-Miller) [reply BEIMA.Backend.Test/DeviceFunctions/AddDeviceTest.cs:134]: Removed them and then re-ordered the mocks so they shouldn't be mixed up anymore.

**Review by josephmoore99 — APPROVED** [review]

**Review by thessbsu — APPROVED** [review]

**Pull request merged**